packages/backend/src/services/PromoteService/PromoteService.ts

```diff
@@ -774,21 +1143,47 @@ export class PromoteService extends BaseService {
                 : [],
         };
 
-        try {
-            PromoteService.checkPromoteDashboardPermissions(
+        return { promotedDashboard, upstreamDashboard };
+    }
+
+    async promoteDashboard(user: SessionUser, dashboardUuid: string) {
+        const dashboard = await this.dashboardModel.getById(dashboardUuid);
+
+        const { upstreamProjectUuid } = await this.projectModel.getSummary(
+            dashboard.projectUuid,
+        );
+        if (!upstreamProjectUuid)
+            throw new NotFoundError(
+                'This chart does not have an upstream project',
+            );
+
+        const { promotedDashboard, upstreamDashboard } =
+            await this.getPromotedDashboard(
+                user,
+                dashboard,
+                upstreamProjectUuid,
+            );
+
+        // We're going to be updating this structure with new UUIDs if we need to create the items (eg: spaces)
+
+        let [promotionChanges, promotedCharts] =
+            await this.getPromotionDashboardChanges(
                 user,
                 promotedDashboard,
                 upstreamDashboard,
             );
 
-            // Check permissions for all chart tiles
+        promotedCharts = [];
```

Comment:
this is for re`setting` the value on the `let` variable. Otherwise, the linter suggests to convert this to a `const`. I could add a comment to skip this validation I guess...